### ISSUE TITLE
Add title to split window

### ISSUE THREAD

**Issue by Jspoor76** [microsoft/terminal#4717]
# Description of the new feature/enhancement

When doing split windows in a tab, there's no header per window, rather the last command given is used int he TAB header. See attached example
![WindowsTerminal](https://user-images.githubusercontent.com/61461157/75248606-c18abe00-57d4-11ea-96f1-9c19bdccbd85.png)
Can you provide per split window a header with the command?

This is when using alt-shift-+ to open e.g. multiple Command prompts to perform pings.

# Proposed technical implementation details (optional)
Add a header to split windows with the command given

**Comment by zadjii-msft** [comment]
Adding a title per-pane is definitely something people will want. I'm gonna throw this on the backlog. 

We might need to clarify what should happen to the tab if there's a titlebar per-pane. Maybe display something like "4 panes"? Or still display the tab title there? THat would lead to just duplicated "cmd - ping 8.8.8.8" text, but that's probably _fine_. We could _probably_ make this a per-profile setting too.

**Comment by DHowett-MSFT** [comment]
Yanking Triage on this one -- we should show something in the title (`cmd (+4 others)`) _and_ above each pane, maybe?

**Comment by jetwiwo** [comment]
> Yanking Triage on this one -- we should show something in the title (`cmd (+4 others)`) _and_ above each pane, maybe?

Yes please. Sometimes I'm running 4 different build windows with similar output that I want to watch. remembering which one is which can be difficult over time. The ability to set a title and view it above each pane would help with this. The "+4 others" would be less important for me personally although I'd like the option to turn it on and off independently of the individual titles if it is included. Perhaps have an option for a shorter version `cmd (+4)` as well for those who don't have much space but want the information.

**Comment by oleg-astakhov** [comment]
From what I understand this has not been implemented yet. I've just started using split window panes of Windows Terminal, and I instantly ran into this issue that it's hard to tell apart which is which.

If you're looking for ideas for how to implement this, then what I was actually looking for is what I've seen in other terminals, which is the following solution:

Allow to set a custom title which would be like a "watermark" behind the printed text, so that the letters would be pretty large, not too bright, not too dark, just enough to be readable. Also, you can let us change their colors. One great example is having multiple panes open for different "environments": PROD, SAND, TEST, DEV and so on. Of course you can label them anything you want, but this just highlights the very obvious usage scenario.

I would very much like this feature to be delivered. I've googled this just in case it's already done, but I couldn't find anything.

**Comment by zadjii-msft** [comment]
@oleg-astakhov sounds kinda like you're asking for something like: #6979.

That's kinda sat in the freezer forever, because it's more or less doable with `backgroundImage` already. Not nearly as flexible as using the title text as the background, but definitely doable today.

For example, I use the following as a "badge" on my sudo profile:

```jsonc
{
    "backgroundImage": "z:\\dev\\public\\sudo\\img\\Windows\\AppList.targetsize-96.png",
    "backgroundImageAlignment": "topRight",
    "backgroundImageOpacity": 0.5,
    "backgroundImageStretchMode": "none",

    "opacity": 85,
    "useAcrylic": true,

    "colorScheme": "Flatland",
    "name": "sudo",
    "tabTitle": "sudo",
    // Other settings omitted...
},
```

![image](https://github.com/user-attachments/assets/4a3976e7-f9fe-456e-ada0-739a7e9b95b6)

**Comment by oleg-astakhov** [comment]
Thanks for the quick response. I I guess it's partially doable with the images, but you got it right, not flexible, especially if I just need to open multiple new panes fast and start working right away. 

For example, in my particular case, I have different docker microservices open in different Tabs and I hop between them. So, I wanted to group them into fewer tabs, and name the panes after the service name. But when I switch to another project, there's a different set of services. So I'd need to name the panes differently.

Creating background images per service, then creating profiles per service, somehow link all that through configuration - that seems too much work and not straightforward. From what I see, the background image is tied to a profile which is tied to a CMD. But I thought I could just add a background for any active pane, without creating a new profile per image, but obviously I can't. 

P.S. I'm new to Windows Terminal, I don't know all the features yet, maybe I'm just missing something... But so far, I couldn't achieve what I need with images. That is, I need different images on different panes but in a single tab, and assign these images on the fly, without pre-configuring profiles beforehand. You know, agile and throw-away style :)

**Comment by mynameistito** [comment]
jump on the list, would love to have this implemented too!

**Comment by ctrl-rohan** [comment]
I see this item has seen some work recently, namely in https://github.com/microsoft/terminal/pull/20068 

By any chance, is this feature available in any preview build? Or is there any way I can generate a version of the program based off https://github.com/sagarbhure-msft/terminal/tree/feature/pane-title ? Very excited for this, thanks for your work.
